lib/dns.js

```diff
@@ -38,6 +37,9 @@ function errnoException(err, syscall, hostname) {
   return ex;
 }
 
+function validIpVersion(isIpResult) {
+  return isIpResult !== 0;
```

Comment:
That's a good point, I still think it helps the code quality but if there is no consensus on that fact I'll do `!== 0` since it's less objectionable and more like the old behavior.